projects/plugins/jetpack/modules/masterbar/admin-menu/load.php

```diff
@@ -86,5 +89,26 @@ function get_admin_menu_class() {
  */
 $admin_menu_class = apply_filters( 'jetpack_admin_menu_class', get_admin_menu_class() );
 if ( should_customize_nav( $admin_menu_class ) ) {
+	/** The admin menu singleton instance. @var Base_Admin_Menu $instance */
 	$admin_menu_class::get_instance();
+
+	/**
+	 * Trigger an event when the user uses the dashboard quick switcher.
+	 *
+	 * @param string $screen The current screen.
+	 * @param string $view The view the user choosed to go to.
+	 */
+	function dashboard_quick_switcher_record_usage( $screen, $view ) {
+		require_once __DIR__ . '/class-dashboard-switcher-tracking.php';
+
+		$tracking = new Dashboard_Switcher_Tracking(
```

Comment:
This part is responsible with loading the implementation that triggers the event. In this manner, Dashboard_Switcher_Tracking is not aware of how we construct the plan or with what configuration we initialize Tracking class.

Reply:
While I appreciate the benefits of decoupling the dependencies and injecting them, I firmly believe this makes the code harder to understand. I'm worried about the future us (or other devs) coming back to this code in order to fix/change something and not being able to do that efficiently because of the code legibility.

Anyway, I know this is a lost fight for me so I'm happy to disagree and commit 🙂.